File: cpp/134_GasStation.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int start = 0;
        int total = 0;
        int remain = 0;
        for(int i = 0; i < gas.size(); ++i){
            remain = remain + gas[i] - cost[i];
            if(remain < 0){
                total += remain;
                start = i+1;
                remain = 0;
            }
        }
        return total + remain < 0 ? -1 : start;
    }
};

File: js/134_GasStation.js
/**
 * @param {number[]} gas
 * @param {number[]} cost
 * @return {number}
 */
var canCompleteCircuit = function(gas, cost) {
    var start = 0;
    var total = 0;
    var remain = 0;
    for(var i = 0; i < gas.length; ++i){
        remain = remain + gas[i] - cost[i];
        if(remain < 0){
            start = i+1;
            total += remain;
            remain = 0;
        }
    }
    return total+remain < 0 ? -1 : start;
};